src/saslclientio.c

```diff
@@ -872,60 +919,64 @@ CONCRETE_IO_HANDLE saslclientio_create(void* io_create_parameters)
     SASL_CLIENT_IO_INSTANCE* result;
 
     /* Codes_SRS_SASLCLIENTIO_01_005: [If xio_create_parameters is NULL, saslclientio_create shall fail and return NULL.] */
-    if ((sasl_client_io_config == NULL) ||
-        /* Codes_SRS_SASLCLIENTIO_01_092: [If any of the sasl_mechanism or underlying_io members of the configuration structure are NULL, saslclientio_create shall fail and return NULL.] */
-        (sasl_client_io_config->underlying_io == NULL) ||
+    if (sasl_client_io_config == NULL)
+    {
+        LogError("NULL io_create_parameters");
+        result = NULL;
+    }
+    /* Codes_SRS_SASLCLIENTIO_01_092: [If any of the sasl_mechanism or underlying_io members of the configuration structure are NULL, saslclientio_create shall fail and return NULL.] */
+    else if ((sasl_client_io_config->underlying_io == NULL) ||
         (sasl_client_io_config->sasl_mechanism == NULL))
     {
+        LogError("Bad parameters: underlying_io = %p, sasl_mechanism = %p",
+            sasl_client_io_config->underlying_io, sasl_client_io_config->sasl_mechanism);
         result = NULL;
     }
     else
     {
         result = malloc(sizeof(SASL_CLIENT_IO_INSTANCE));
         /* Codes_SRS_SASLCLIENTIO_01_006: [If memory cannot be allocated for the new instance, saslclientio_create shall fail and return NULL.] */
-        if (result != NULL)
+        if (result == NULL)
+        {
+            LogError("Cannot allocate sasl client IO instance");
+        }
+        else
         {
             result->underlying_io = sasl_client_io_config->underlying_io;
-            if (result->underlying_io == NULL)
+            /* Codes_SRS_SASLCLIENTIO_01_089: [saslclientio_create shall create a frame_codec to be used for encoding/decoding frames bycalling frame_codec_create and passing the underlying_io as argument.] */
+            result->frame_codec = frame_codec_create(on_frame_codec_error, result);
+            if (result->frame_codec == NULL)
             {
+                /* Codes_SRS_SASLCLIENTIO_01_090: [If frame_codec_create fails, then saslclientio_create shall fail and return NULL.] */
+                LogError("frame_codec_create failed");
                 free(result);
                 result = NULL;
             }
             else
             {
-                /* Codes_SRS_SASLCLIENTIO_01_089: [saslclientio_create shall create a frame_codec to be used for encoding/decoding frames bycalling frame_codec_create and passing the underlying_io as argument.] */
-                result->frame_codec = frame_codec_create(on_frame_codec_error, result);
-                if (result->frame_codec == NULL)
+                /* Codes_SRS_SASLCLIENTIO_01_084: [saslclientio_create shall create a sasl_frame_codec to be used for SASL frame encoding/decoding by calling sasl_frame_codec_create and passing the just created frame_codec as argument.] */
```

Comment:
Was this updated on the requirements doc as well? I don't see it included in this PR.